spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessorTests.java

```diff
@@ -273,6 +278,42 @@ public void queuesToDeclare() {
 		context.close();
 	}
 
+	@Test
+	public void concurrency() throws InterruptedException, ExecutionException {
+		final int concurrencyLevel = 8;
+		final ExecutorService executorService = Executors.newFixedThreadPool(concurrencyLevel);
+		try {
+			for (int i = 0; i < 1000; ++i) {
```

Comment:
Since we test in this method really a concurrency of the `RabbitListenerAnnotationBeanPostProcessor`, there is no better place unless this test class.
I'll review the test locally thoroughly for performance.